node/network/bridge/src/action.rs

```diff
@@ -76,21 +80,42 @@ pub(crate) enum Action {
 		Vec<WireMessage<protocol_v1::CollationProtocol>>,
 	),
 
-	Abort,
+	/// Send a message to another subsystem or the overseer.
+	///
+	/// Used for handling incoming requests.
+	SendMessage(AllMessages),
+
+	/// Abort with reason.
+	Abort(AbortReason),
 	Nop,
 }
 
+#[derive(Debug)]
+pub(crate) enum AbortReason {
```

Comment:
Do we want this to be an error type as well? It would also allow to backtrace the abort reason backtrace on reception in debug mode i.e.

Reply:
Hmm, not every AbortReason is an error though.

Reply:
Given that you `match` on this enum, this feature is already there :)

Reply:
It won't prevent you from matching on the error variants, it just would seem somewhat natural to me. But I frankly have no strong opinion about this.

Reply:
True, but it feels wrong to derive `Error` for something that is not. It is just a reason for abort, some of the constructors of course are some variants of unexpected `Abort`. I could remove the expected abort type from the enum and instead use something like  `Result<(), ErrorAbort>`, but I think I would prefer postponing that until the need arises.